src/contract/assignments.rs

```diff
@@ -622,6 +647,58 @@ impl ConcealSeals for Assignments {
     }
 }
 
+impl MergeRevealed for Assignments {
+    fn merge_revealed_with(&self, other: &Self) -> Result<Self, String> {
+        // Better ways to do this?
+        if self.consensus_commitments() == other.consensus_commitments() {
+            match (self, other) {
+                (
+                    Assignments::Declarative(first_vec),
+                    Assignments::Declarative(second_vec),
+                ) => {
+                    let mut result = vec![];
+                    for (first, second) in
+                        first_vec.iter().zip(second_vec.iter())
+                    {
+                        result.push(first.merge_revealed_with(second)?);
+                    }
+                    Ok(Assignments::Declarative(result))
+                }
+
+                (
+                    Assignments::DiscreteFiniteField(first_vec),
+                    Assignments::DiscreteFiniteField(second_vec),
+                ) => {
+                    let mut result = vec![];
+                    for (first, second) in
+                        first_vec.iter().zip(second_vec.iter())
+                    {
+                        result.push(first.merge_revealed_with(second)?);
+                    }
+                    Ok(Assignments::DiscreteFiniteField(result))
+                }
+
+                (
+                    Assignments::CustomData(first_vec),
+                    Assignments::CustomData(second_vec),
+                ) => {
+                    let mut result = vec![];
+                    for (first, second) in
+                        first_vec.iter().zip(second_vec.iter())
+                    {
+                        result.push(first.merge_revealed_with(second)?);
+                    }
+                    Ok(Assignments::CustomData(result))
+                }
+
+                _ => Err(s!("Assignments type mismatch in merge operation")),
```

Comment:
This should be marked as `unreachable!` instead of error: this is a HUGE BUG if two items with the same commitment ID have different state and the whole app should crash to attract attention to this fact.

Reply:
Pattern mismatches are marked as unreachable in the next update.